services/provider/server/server_test.go

```diff
@@ -0,0 +1,298 @@
+				},
+			},
+		},
+	}
+
+	for _, i := range consumerResource.Status.CephResources {
+		if i.Kind == "CephClient" {
+			cephClient, secret := createCephClientAndSecret(i.Name, server)
+			cephClient.Status = &rookCephv1.CephClientStatus{
+				Info: map[string]string{
+					"secretName": fmt.Sprintf("rook-ceph-client-%s", i.Name),
+				},
+			}
+
+			assert.NoError(t, client.Create(ctx, cephClient))
+			assert.NoError(t, client.Create(ctx, secret))
+		}
+	}
+
+	monCm, monSc, mgrSvc := createMonConfigMapandSecret(server)
+	assert.NoError(t, client.Create(ctx, monCm))
+	assert.NoError(t, client.Create(ctx, monSc))
+	assert.NoError(t, client.Create(ctx, mgrSvc))
+
+	exR, err := server.getExternalResources(ctx, consumerResource)
+	assert.NoError(t, err)
+	for i := range mockExtR {
+		data, err := json.Marshal(mockExtR[i].Data)
+		assert.NoError(t, err)
+		assert.Equal(t, string(exR[i].Data), string(data))
+		assert.Equal(t, exR[i].Name, mockExtR[i].Name)
+		assert.Equal(t, exR[i].Kind, mockExtR[i].Kind)
+	}
+
+	// When CephClient status is empty
+
+	object = []runtime.Object{
+		&rookCephv1.CephClient{},
+	}
+	s = runtime.NewScheme()
+	assert.NoError(t, v1.AddToScheme(s))
+	client = newFakeClient(t, object...)
+	server = &OCSProviderServer{
+		client:    client,
+		namespace: "openshift-storage",
+	}
+
+	for _, i := range consumerResource.Status.CephResources {
+		if i.Kind == "CephClient" {
+			cephClient, secret := createCephClientAndSecret(i.Name, server)
+			assert.NoError(t, client.Create(ctx, cephClient))
+			assert.NoError(t, client.Create(ctx, secret))
+
+		}
+	}
+
+	exR, err = server.getExternalResources(ctx, consumerResource)
+	assert.Error(t, err)
+	assert.NotEqual(t, len(mockExtR), len(exR))
+
+	// When CephClient status info is empty
+
+	object = []runtime.Object{
+		&rookCephv1.CephClient{},
+	}
+	s = runtime.NewScheme()
+	assert.NoError(t, v1.AddToScheme(s))
+	client = newFakeClient(t, object...)
+	server = &OCSProviderServer{
+		client:    client,
+		namespace: "openshift-storage",
+	}
+
+	for _, i := range consumerResource.Status.CephResources {
+		if i.Kind == "CephClient" {
+			cephClient, secret := createCephClientAndSecret(i.Name, server)
+			cephClient.Status = &rookCephv1.CephClientStatus{
+				Info: map[string]string{},
+			}
+
+			assert.NoError(t, client.Create(ctx, cephClient))
+			assert.NoError(t, client.Create(ctx, secret))
+
+		}
+	}
+
+	monCm, monSc, mgrSvc = createMonConfigMapandSecret(server)
+	assert.NoError(t, client.Create(ctx, monCm))
+	assert.NoError(t, client.Create(ctx, monSc))
+	assert.NoError(t, client.Create(ctx, mgrSvc))
+
+	exR, err = server.getExternalResources(ctx, consumerResource)
+	assert.Equal(t, err, fmt.Errorf("cephClient cephuser-rbd-node Status.Info is empty"))
+	assert.NotEqual(t, len(mockExtR), len(exR))
+
+	// when storageConsume status is empty
+	exR, err = server.getExternalResources(ctx, &ocsv1alpha1.StorageConsumer{})
+	assert.Equal(t, err, fmt.Errorf("ceph resources are empty"))
```

Comment:
The check here is incorrect, you are testing the API server, not the storage consumer controller. 
The API server does not have bais toward the number (or even existence) of resources in the StorageConsumer status as such the return value here will not be an error, it will be an empty array.